src/v2/index.ts

```diff
@@ -54,14 +52,13 @@ export interface LanguageResult {
 
 export interface GetLanguagesCallback {
   (err: Error|null, results?: LanguageResult[]|null,
-   apiResponse?: r.Response): void;
+   apiResponse?: Metadata): void;
 }
 
 export interface TranslateConfig extends GoogleAuthOptions {
   key?: string;
   autoRetry?: boolean;
   maxRetries?: number;
-  request?: typeof r;
```

Comment:
@JustinBeckwith is this the thing you meant we removed that shouldn't have been there? Did `request` ever actually exist on `GoogleAuthOptions`? I don't see it here: https://github.com/googleapis/google-auth-library-nodejs/blob/37bb8c7cd0a6501103274284d9cddd6816cc881e/src/auth/googleauth.ts

If it was just a pretend setting, then I would say non-breaking is okay, since it never had functionality anyway. Sorry if I missed something!

Reply:
Aye, that was the thing.  This allowed the user to technically override the implementation of the `request` module used.  We did this in other modules in a way that was totally internal, but for some reason it was exposed here publicly.  Theoretically - someone could have been passing in the full `request` module here, hoping to use that instead of `teeny-request`.  It's a stretch.

Reply:
*Sorry in advance, because I'm trying to follow along with the history of the changes*

We removed the `requestModule` option from common a couple of months ago, which was a breaking change. So did this library actually intend to support the feature, but we forgot to remove the code here?

Here's the code as it is today in this repo:

```js
const config = {
  baseUrl,
  scopes: ['https://www.googleapis.com/auth/cloud-platform'],
  packageJson: require('../../../package.json'),
  projectIdRequired: false,
  requestModule: teenyRequest as typeof r, // impenetrable! 
};

super(config, options);
```

So, while the user's `options.request` is passed to `Service`, it would have always been ignored in favor of the `config.requestModule` that we set. Here *was* the code in common, before we deleted it ([source](https://github.com/googleapis/nodejs-common/pull/394/files#diff-22ccf3ebb0cb6b300ee90a38b88edff8L102)):

```js
this.requestModule = config.requestModule;

const reqCfg = extend({}, config, {
  projectIdRequired: this.projectIdRequired,
  projectId: this.projectId,
  credentials: options.credentials,
  keyFile: options.keyFilename,
  email: options.email,
  token: options.token,
  request: this.requestModule,
  token: options.token
});
```

Admittedly, trying to follow along with `request` vs `requestModule` and `options` vs `config` and the ordering of assignments in the constructor of this module creates loopholes that I can't really find my way out of. But to me, it seems like even though we said we accepted `options.request`, it never did anything anyway-- especially since even if we had all of our settings correct, it wouldn't have made a difference since the feature was killed upstream in common.